doc/source/cluster/kubernetes/user-guides/kuberay-gcs-persistent-ft.md

```diff
@@ -0,0 +1,150 @@
+(AOF) log. When using using these options, it is recommended to disable AOF.
+
+With GCP GKE and Azure AKS, the default storage classes are [persistent
+disks](https://cloud.google.com/kubernetes-engine/docs/concepts/persistent-volumes)
+and [SSD Azure
+disks](https://learn.microsoft.com/en-us/azure/aks/azure-csi-disk-storage-provision)
+respectively, and the only configuration needed to provision a disk is as
+follows:
+
+```
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: redis-data
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 8Gi
+  storageClassName: standard-rwo
+```
+
+On AWS, you must [Create a storage
+class](https://docs.aws.amazon.com/eks/latest/userguide/create-storage-class.html)
+yourself as well.
+
+## Tuning backups
+
+Redis supports database dumps at set intervals, which is good for fast recovery
+and high performance during normal operation.
+
+Redis also supports journaling at frequent intervals (or continuously), which
+can provide stronger durabililty at the cost of more disk writes (i.e., slower
+performance).
+
+A good starting point for backups is to enable both, which can be done like so:
+
+```
+# Dump a backup every 60s, if there are 1000 writes since the prev. backup.
+save 60 1000
+dbfilename dump.rdb
+
+# Enable the append-only log file.
+appendonly yes
+appendfilename "appendonly.aof"
+
+```
+
+In this recommended configuration, full backups are created every 60s while the
+append-only log is updated every second, which is a reasonable balance for disk
+space, latency, and data safety.
+
+There are more options to configure the AOF, defaults shown here:
+
+```
+# Sync the log to disk every second.
+# Alternatives are "no" and "always" (every write).
+appendfsync everysec
+auto-aof-rewrite-percentage 100
+auto-aof-rewrite-min-size 64mb
+```
+
+You can view the full reference
+[here](https://raw.githubusercontent.com/redis/redis/refs/tags/7.4.0/redis.conf).
+
+
+If your job is generally idempotent and can resume from several minutes of state
+loss, you may prefer to disable the append-only log.
+
+If you prefer your job to lose as little state as possible, then you may prefer
+to set `appendfsync` to `always`, such that all write are stored immediately.
+
+## Putting it together
+
+Edit [the full
+YAML](https://github.com/ray-project/kuberay/blob/master/config/samples/ray-cluster.persistent-redis.yaml)
+to your satisfaction and apply it:
+
+```
+kubectl apply -f config/samples/ray-cluster.persistent-redis.yaml
+```
+
+Verify that a disk has been provisioned and redis is running:
+
+```
+kubectl get persistentvolumes
+kubectl get pods
+# Should see redis-0 running.
+```
+
+After running a job with some state in GCS, you will be able to delete the ray
+head pod as well as the redis pod without data loss.
+
+## Verifying
+
+Forward connections to the ray cluster you just created with the Ray Kubectl
```

Comment:
hyper link Ray Kubectl plugin to https://docs.ray.io/en/latest/cluster/kubernetes/user-guides/kubectl-plugin.html